File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/RxBusDemoActivity.java
package io.ganguo.demo.view.activity;

import android.content.Context;
import android.content.Intent;

import io.ganguo.demo.databinding.ActivityRxbusDemoBinding;
import io.ganguo.demo.view.activity.base.GGVModelActivity;
import io.ganguo.demo.viewmodel.activity.RxBusDemoVModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.vmodel.ViewModelHelper;

/**
 * RxBusDemo
 * Created by leo on 2018/8/15.
 */
public class RxBusDemoActivity extends GGVModelActivity<ActivityRxbusDemoBinding, RxBusDemoVModel> {
    public static Intent intentFor(Context context) {
        return new Intent(context, RxBusDemoActivity.class);
    }

    @Override
    public RxBusDemoVModel createViewModel() {
        return new RxBusDemoVModel();
    }

    @Override
    public void onViewAttached(RxBusDemoVModel viewModel) {
        ViewModelHelper.bind(getBinding().includeHeader, viewModel, onCreateHeaderVModel());
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/RxBusDemoActivity.kt
package io.ganguo.demo.view.activity

import android.content.Context
import android.content.Intent
import io.ganguo.demo.databinding.ActivityRxbusDemoBinding
import io.ganguo.demo.view.activity.base.GGVModelActivity
import io.ganguo.demo.viewmodel.activity.RxBusDemoVModel
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.pack.common.HeaderViewModel
import io.ganguo.viewmodel.core.ViewModelHelper

/**
 * RxBusDemo
 * Created by leo on 2018/8/15.
 */
class RxBusDemoActivity : GGVModelActivity<ActivityRxbusDemoBinding, RxBusDemoVModel>() {

    override fun createViewModel(): RxBusDemoVModel {
        return RxBusDemoVModel()
    }

    override fun onViewAttached(viewModel: RxBusDemoVModel) {
        ViewModelHelper.bind(binding.includeHeader, viewModel, newHeaderVModel())
    }


    companion object {
        fun intentFor(context: Context): Intent {
            return Intent(context, RxBusDemoActivity::class.java)
        }
    }

}
